coriolis/tests/scheduler/filters/test_trivial_filters.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2023 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+from unittest import mock
+
+import ddt
+
+from coriolis.scheduler.filters import trivial_filters
+from coriolis.tests import test_base
+
+
+@ddt.ddt
+class RegionsFilterTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the RegionsFilter class."""
+
+    def setUp(self):
+        super(RegionsFilterTestCase, self).setUp()
+        self.regions = [mock.sentinel.region1, mock.sentinel.region2]
+        self.any_region = False
+        self.regions_filter = trivial_filters.RegionsFilter(
+            self.regions, self.any_region)
+
+    def test__repr__(self):
+        result = repr(self.regions_filter)
+
+        self.assertEqual(result, "<RegionsFilter(regions=%s, "
+                                 "any_region=%s)>" % (self.regions,
+                                                      self.any_region))
+
+    @ddt.data(
+        (None, [mock.sentinel.region1, mock.sentinel.region2], 100),
+        ([mock.sentinel.region1, mock.sentinel.region2],
+            [mock.sentinel.region1, mock.sentinel.region2], 100),
+        ([mock.sentinel.region1, mock.sentinel.region2], [], 0),
+        ([mock.sentinel.region1, mock.sentinel.region2],
+            [mock.sentinel.region1], 0)
+    )
+    @ddt.unpack
+    def test_rate_service(self, regions, mapped_regions, expected):
+        self.regions_filter._regions = regions
+        mock_service = mock.Mock()
+        mock_service.mapped_regions = [mock.Mock(id=region) for region in
+                                       mapped_regions]
+
+        result = self.regions_filter.rate_service(mock_service)
+
+        self.assertEqual(result, expected)
```

Comment:
There's a missed case here. Please account for the case where `any_regions` is set to `True`, and also at least one (not all) of the regions are found in `mapped_regions` (just like the last test case).